lib/teamdynamix_api.rb

```diff
@@ -93,16 +98,21 @@ def valid_auth_token?(token)
     token.match(/^[a-zA-Z0-9\.\-\_]*$/)
   end
 
+  def retire_asset_payload(asset_id)
+    asset = get_asset(asset_id)
+    asset.merge!(API_CONFIG[:delete].stringify_keys)
```

Comment:
Hmm, I think the yaml parser can be configured to stringify the keys or switch to hash with indeffernent access. Does the foreman config loader not do this? Did you actually have an issue here?

Reply:
asset returned has stringified keys so in order to overwrite the StatusID I am stringifying that key along with any other attributes user may have configured to be updated on retire.

is there something wrong with using stringify_keys?

Reply:
No, I'm just wondering if Foreman Core is already doing it or similar.
Did you inspect to make sure its not already handled?